api/src/main/java/javax/config/spi/ConfigSource.java

```diff
@@ -85,21 +86,24 @@
     int DEFAULT_ORDINAL = 100;
 
     /**
-     * Return the properties in this config source
-     * @return the map containing the properties in this config source
+     * Return the properties in this config source.
+     *
+     * @return the map containing the properties in this config source if these can be scanned, empty if config source cannot
+     * read the config keys and values (due to performance or feature set reasons)
      */
-    Map<String, String> getProperties();
```

Comment:
We had this information in DeltaSpike. But now with an Optional but with a specific method. Could now be a default method of course:
https://github.com/apache/deltaspike/blob/master/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/spi/config/ConfigSource.java#L103
This flag basically indicates that the `getProperties()` method is either slow or incomplete.
It was even part of the initial proposal in July 2016
https://github.com/struberg/javaConfig/blob/cf41cf130bcaf5447ff8781df8f845bab41cb5b2/api/src/main/java/javax/config/spi/ConfigSource.java#L94

But it turned out that this information is pretty much useless!
In which situation would you use it? We never had that case so far. The closest use case is to warn the user that a specific ConfigSource might be slow. Well, that's not much we gained. And that was the reason why I left it off for the ConfigJSR/mp-config:
https://github.com/struberg/javaConfig/commit/2bf96640d8f1f75f2163c85daf36e3d20508f0f6

Reply:
Back to original, with isScannable added from DeltaSpike by @struberg